src/Shared/UnitTests/FileUtilities_Tests.cs

```diff
@@ -878,5 +837,37 @@ public void PathDoesNotLookLikeUnixPathOnWindows()
 
         private static string SystemSpecificAbsolutePath => FileUtilities.ExecutingAssemblyPath;
 
+
+        [Fact]
+        public void GetFolderAboveTest()
+        {
+            string root = NativeMethodsShared.IsWindows ? @"c:\" : "/";
+            string path = Path.Combine(root, "1", "2", "3", "4", "5");
+
+            Assert.Equal(Path.Combine(root, "1", "2", "3", "4", "5"), FileUtilities.GetFolderAbove(path, 0));
+            Assert.Equal(Path.Combine(root, "1", "2", "3", "4"), FileUtilities.GetFolderAbove(path));
+            Assert.Equal(Path.Combine(root, "1", "2", "3"), FileUtilities.GetFolderAbove(path, 2));
+            Assert.Equal(Path.Combine(root, "1", "2"), FileUtilities.GetFolderAbove(path, 3));
+            Assert.Equal(Path.Combine(root, "1"), FileUtilities.GetFolderAbove(path, 4));
```

Comment:
This was done with literal paths in master but this should be equivalent and slash-direction agnostic.